roles/openshift_cluster_monitoring_operator/defaults/main.yml

```diff
@@ -21,16 +21,16 @@ l_openshift_cluster_monitoring_image_dicts:
-    node_exporter: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'prometheus-node-exporter') }}"
-    prometheus_config_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'prometheus-config-reloader') }}"
-    configmap_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'configmap-reloader') }}"
-    grafana: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'grafana') }}"
-    kube_state_metrics: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'kube-state-metrics') }}"
-    kube_rbac_proxy: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}:${version}' | regex_escape, 'kube-rbac-proxy') }}"
-    oauth_proxy: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}:${version}' | regex_escape, 'oauth-proxy') }}"
+    prometheus_operator: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}' | regex_escape, 'prometheus-operator') | regex_replace(':[^:]*$' | regex_escape, '') }}"
+    prometheus: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}' | regex_escape, 'prometheus') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    alertmanager: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}' | regex_escape, 'prometheus-alertmanager') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    node_exporter: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}' | regex_escape, 'prometheus-node-exporter') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    prometheus_config_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}' | regex_escape, 'prometheus-config-reloader') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    configmap_reloader: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}' | regex_escape, 'configmap-reloader') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    grafana: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('ose-${component}' | regex_escape, 'grafana') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    kube_state_metrics: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}' | regex_escape, 'kube-state-metrics') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
+    kube_rbac_proxy: "{{ l_openshift_cluster_monitoring_non_standard_reg_url | regex_replace('${component}' | regex_escape, 'kube-rbac-proxy') | regex_replace(':[^:]*$' | regex_escape, '')  }}"
```

Comment:
prometheus_operator, configmap_reloader, prometheus_config_reloader, kube_state_metrics, and kube_rbac_proxy differ from the others in that they don't replace the 'ose-' prefix too. This isn't a change introduced in this PR, just wanted to mention it in case the original state was incorrect.

Reply:
This is correct, for historic reasons the other images don't have the `ose-` prefix.